lib/oper1.g

```diff
@@ -144,21 +144,26 @@ fi;
 ##
 BIND_GLOBAL( "INSTALL_METHOD_FLAGS",
     function( opr, info, rel, flags, rank, method )
-    local   methods,  narg,  i,  k,  tmp, replace, match, j, lk;
+    local   methods,  narg,  i,  k,  tmp, replace, match, j, lk, ranknow;
 
     if IsHPCGAP then
         # TODO: once the GAP compiler supports 'atomic', use that
         # to replace the explicit locking and unlocking here.
         lk := WRITE_LOCK(METHODS_OPERATION_REGION);
     fi;
     # add the number of filters required for each argument
+    if IS_FUNCTION(rank) then
+        ranknow := rank();
+    else 
+        ranknow := rank;
+    fi;    
```

Comment:
Perhaps rename rank to baserank, and ranknow to rank, in order to reduce the diffs?